src/main/java/com/direwolf20/buildinggadgets/common/items/gadgets/GadgetDestruction.java

```diff
@@ -299,34 +290,7 @@ public void clearArea(World world, BlockPos pos, Direction side, ServerPlayerEnt
             player.sendMessage(TooltipTranslation.GADGET_PALETTE_OVERFLOW.componentTranslation());
             return;
         }
-
-        WorldSave worldSave = WorldSave.getWorldSaveDestruction(world);
-        worldSave.addToMap(getUUID(stack).toString(), snapshot.serialize());
-    }
-
-    public static void undo(PlayerEntity player, ItemStack stack) {
-        World world = player.world;
-        WorldSave worldSave = WorldSave.getWorldSaveDestruction(world);
-
-        CompoundNBT serializedSnapshot = worldSave.getCompoundFromUUID(getUUID(stack).toString());
-        if (serializedSnapshot.isEmpty())
-            return;
-
-        RegionSnapshot snapshot = RegionSnapshot.deserialize(serializedSnapshot);
-        restoreSnapshotWithBuilder(world, snapshot);
-        worldSave.addToMap(getUUID(stack).toString(), new CompoundNBT());
-        worldSave.markDirty();
-    }
-
-    public static void restoreSnapshotWithBuilder(World world, RegionSnapshot snapshot) {
-        Set<BlockPos> pastePositions = snapshot.getTileData().stream()
-                .map(Pair::getLeft)
-                .collect(Collectors.toSet());
-        int index = 0;
-        for (BlockPos pos : snapshot.getPositions()) {
-            snapshot.getBlockStates().get(index).ifPresent(state -> EffectBlock.spawnEffectBlock(world, pos, new BlockData(state, TileSupport.dummyTileEntityData()), EffectBlock.Mode.PLACE, pastePositions.contains(pos)));
-            index++;
-        }
+        addUndo(stack, snapshot);
```

Comment:
Can we rename this method to `addUndoToStack` or pushUndoToStack` add undo just seems a bit ominous :P